threshold-network/token-dashboard
# Upgrade flow

Depends on: https://github.com/threshold-network/token-dashboard/tree/erc20-approvals

WIP...

To test the dapp locally:

* run your local chain eg with Ganache
* clone https://github.com/keep-network/keep-core repo:
** go to `/solidity-v1` dir
** run `npm install`
** run `npx truffle migrate --reset --network local`- make sure you use the same port as in [`/solidity-v1/truffle-config.js`](https://github.com/keep-network/keep-core/blob/main/solidity-v1/truffle-config.js#L8-L11)
** run `rm -rf artifacts` and `cp -r build/contracts artifacts`
** link the package with `yarn link` in `/solidity-v1`
* clone https://github.com/threshold-network/solidity-contracts repo and checkout to `deploy-vending-machine-for-nu` branch
** run `yarn`
** link local `@keep-network/keep-core` package via `yarn link @keep-network/keep-core`
** deploy contracts via `yarn deploy --network development --reset`- make sure you use the same port as in [`hardhat.config.ts`](https://github.com/threshold-network/solidity-contracts/blob/main/hardhat.config.ts#L42-L44)
** run `./scripts/prepare-artifacts.sh --network development`
** link the package with `yarn link`
* install `token-dashboard` dependencies
* link `@keep-network/keep-core` and `@threshold-network/solidity-contracts` via `yarn link @keep-network/keep-core @threshold-network/solidity-contracts`
* start the dapp `yarn start`
